solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesListItem/PRAndIssuesListItem.tsx

```diff
@@ -9,20 +9,37 @@ import {
   CommentIcon,
   ClosedIssueIcon,
 } from '../Icons';
-import { type } from '../PRAndIssuesHeader';
-import { Issue } from '~/types/issues-type';
-
-const PRAndIssuesListItem = (props: Issue) => {
-  const [local] = splitProps(props, [
-    'number',
-    'title',
-    'url',
-    'state',
-    'createdAt',
-    'login',
-    'commentCount',
-    'labels',
-  ]);
+import { IssueNodeProps } from '~/types/issues-type';
+import { PullRequestNodeProps } from '~/types/pull-request-type';
+import { Label } from '~/types/label-type';
+
+interface PRAndIssuesListItemProps {
+  issue?: IssueNodeProps;
+  pullRequest?: PullRequestNodeProps;
+}
+
+const PRAndIssuesListItem = (props: PRAndIssuesListItemProps) => {
+  const [local] = splitProps(props, ['issue', 'pullRequest']);
+
+  const type = local.issue ? 'issue' : 'pr';
+
+  const state = local.issue ? local.issue.state : local?.pullRequest?.state;
+
+  const url = local.issue?.url || local.pullRequest?.url;
+
+  const title = local.issue?.title || local.pullRequest?.title;
+
+  const number = local.issue?.number || local.pullRequest?.number;
+
+  const labels = local.issue?.labels || local.pullRequest?.labels || [];
+
+  const login = local.issue?.author?.login || local.pullRequest?.author?.login;
+
+  const createdAt = local.issue?.createdAt || local.pullRequest?.createdAt;
+
+  const commentCount =
+    local.issue?.comments?.totalCount ||
+    local.pullRequest?.comments?.totalCount;
```

Comment:
We should find a better way to manage all this, because this is a solid anti-pattern